components/scream/src/physics/p3/p3_inputs_initializer.cpp

```diff
@@ -117,6 +124,12 @@ void P3InputsInitializer::initialize_fields ()
   auto h_nccn_prescribed = m_fields.at("nccn_prescribed").get_reshaped_view<Pack**,Host>();
   auto h_inv_qc_relvar   = m_fields.at("inv_qc_relvar").get_reshaped_view<Pack**,Host>();
 
+  // dp might be inited by Homme.
+  decltype(h_pmid) h_dp;
+  if (m_fields.count("dp")==1) {
+    h_dp = m_fields.at("dp").get_reshaped_view<Pack**,Host>();
+  }
+
```

Comment:
So, presumably, a subsequent PR will get rid of this whole paradigm right?  And we won't have `p3_inputs_initializer` at all.

Reply:
Right. I'm already working on a branch. Purging all those files/lines feels so good. :)